src/jit/codegenlegacy.cpp

```diff
@@ -21307,6 +21310,21 @@ regNumber CodeGen::genPInvokeCallProlog(LclVarDsc*            frameListRoot,
 
     noway_assert(compiler->lvaInlinedPInvokeFrameVar != BAD_VAR_NUM);
 
+#ifdef _ARM_
+    if (compiler->opts.ShouldUsePInvokeHelpers())
+    {
+        regNumber baseReg;
+        int       adr = compiler->lvaFrameAddress(compiler->lvaInlinedPInvokeFrameVar, true, &baseReg, 0);
```

Comment:
@jkotas You're right. I moved call of CORINFO_HELP_JIT_PINVOKE_BEGIN upper

the generated code is now
```
Instruction list before instruction issue System.Runtime.InternalCalls:RhpCollect(int,int):          master [fa0ef109f] modified untracked

G_M186_IG01:        ; func=00, offs=000000H, size=000EH, gcrefRegs=0000 {}, byrefRegs=0000 {}, byref, nogc <-- Prolog IG

000000      push    {r4,r5,r6,r7,r8,r9,r10,r11,lr}
000004      sub     sp, 36
000006      add     r11, sp, 64
00000A      str     r0, [sp+0x1c]       // [V00 arg0]
00000C      str     r1, [sp+0x18]       // [V01 arg1]

G_M186_IG02:        ; offs=00000EH, size=001EH, gcrefRegs=0000 {}, byrefRegs=0000 {}, byref

00000E      sub     r0, r11, 76
000012      movw    r3, LOW RELOC 0x422ef0
000016      movt    r3, HIGH RELOC 0x422ef0
00001A      blx     r3          // CORINFO_HELP_JIT_PINVOKE_BEGIN
00001C      ldr     r0, [sp+0x1c]       // [V00 arg0]
00001E      ldr     r1, [sp+0x18]       // [V01 arg1]
000020      movw    r3, LOW RELOC 0x423278
000024      movt    r3, HIGH RELOC 0x423278
000028      ldr     r3, [r3]
00002A      blx     r3          // System.Runtime.InternalCalls:RhpCollect(int,int)

G_M186_IG03:        ; offs=00002CH, size=0010H, gcrefRegs=0000 {}, byrefRegs=0000 {}, byref, isz

00002C      sub     r0, r11, 76
000030      movw    r3, LOW RELOC 0x422f00
000034      movt    r3, HIGH RELOC 0x422f00
000038      blx     r3          // CORINFO_HELP_JIT_PINVOKE_END
00003A      b       SHORT G_M186_IG04

G_M186_IG04:        ; offs=00003CH, size=0006H, gcrefRegs=0000 {}, byrefRegs=0000 {}, byref, epilog, nogc

00003C      add     sp, 36
00003E      pop     {r4,r5,r6,r7,r8,r9,r10,r11,pc}
```